cpp/cmake_modules/DefineOptions.cmake

```diff
@@ -470,6 +470,16 @@ takes precedence over ccache if a storage backend is configured" ON)
                 "Rely on jemalloc shared libraries where relevant"
                 ${ARROW_DEPENDENCY_USE_SHARED})
 
+  if(MSVC)
+    # It seems that conda's llvmdev for Windows doesn't provide shared
+    # library.
+    set(ARROW_LLVM_USE_SHARED_DEFAULT OFF)
```

Comment:
Hmm, I don't think we should change this for conda. Instead, conda users on Windows will have to explicitly pass `-DARROW_LLVM_USE_SHARED=OFF`.

Reply:
It seems that LLVM itself doesn't support shared library on Windows: https://github.com/llvm/llvm-project/blob/main/llvm/tools/llvm-shlib/CMakeLists.txt#L14-L16